Nimble/Matchers/MatcherProtocols.swift

```diff
@@ -38,8 +38,33 @@ extension NSArray : NMBOrderedCollection {}
 /// Protocol for types to support beCloseTo() matcher
 @objc public protocol NMBDoubleConvertible {
     var doubleValue: CDouble { get }
+    var stringRepresentation: String { get }
+}
+extension NSNumber : NMBDoubleConvertible {
+    public var stringRepresentation: String {
+        get {
+            return NSString(format: "%.4f", (self)).description
+        }
+    }
+}
+
+private let dateFormatter = NSDateFormatter()
+
+extension NSDate: NMBDoubleConvertible {
+    public var doubleValue: CDouble {
+        get {
+            return self.timeIntervalSinceReferenceDate
+        }
+    }
+    
+    public var stringRepresentation: String {
+        get {
+            dateFormatter.dateFormat = "yyyy-MM-dd HH:mm:ss.SSSS"
```

Comment:
Setting the dateFormatter's `dateFormat` property generally negates the performance benefits of memoizing the instance. Although that behavior may have changes in later versions of iOS, I haven't verified recently.

You can either inline the `let` expression or use an anonymous closure invocation:

```swift
private let dateFormatter = {
    let formatter = NSDateFormatter()
    // ...
   return formatter
}()
```